### ISSUE TITLE
numpy.ndarray size incompatibility in s3prl

### ISSUE THREAD

**Issue by jimbozhang** [espnet/espnet#4019]
**Describe the bug**
I tried to decode with the pretrained `simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp` model, but got an error which indicated that there is a numpy incompatibility.

**Basic environments:**
 - OS information: Linux 5.4.0-91-generic 102-Ubuntu SMP Fri Nov 5 16:31:28 UTC 2021 x86_64
- python version: `3.9.5 (default, Nov 23 2021, 15:27:38)  [GCC 9.3.0]`
- espnet version: `espnet 0.10.6a1`
- Git hash: `48422215e272812feb9bbac9d7cf4aae6a316bca`
  - Commit date: `Fri Jan 28 17:25:31 2022 +0000`
- pytorch version: `pytorch 1.10.2+cu102`

**Environments from `torch.utils.collect_env`:**
PyTorch version: 1.10.2+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Linux Mint 20.2 (x86_64)
GCC version: (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.9.5 (default, Nov 23 2021, 15:27:38)  [GCC 9.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-91-generic-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] pytorch-lightning==1.5.9
[pip3] pytorch-ranger==0.1.1
[pip3] pytorch-wpe==0.0.1
[pip3] torch==1.10.2
[pip3] torch-complex==0.2.1
[pip3] torch-optimizer==0.3.0
[pip3] torch-stoi==0.1.2
[pip3] torchaudio==0.10.2
[pip3] torchmetrics==0.7.0
[pip3] torchvision==0.11.3
[conda] Could not collect

**Task information:**
 - Task: ASR
 - Recipe: librispeech
 - ESPnet2

**To Reproduce**
```
# install s3prl
cd tools
installers/install_s3prl.sh
pip install -e .
cd -

# inference using the pretrain model from HuggingFace
# https://huggingface.co/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp
cd egs2/librispeech/asr1
./run.sh --skip_data_prep false --skip_train true --download_model espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp
```

**Error logs**
`espnet/egs2/librispeech/asr1/exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/decode_asr_lm_lm_train_lm_transformer2_en_bpe5000_valid.loss.ave_asr_model_valid.ac
c.ave/test_clean/logdir/asr_inference.1.log`
```
# python3 -m espnet2.bin.asr_inference --batch_size 1 --ngpu 0 --data_path_and_name_and_type dump/raw/test_clean/wav.scp,speech,sound --key_file exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/decode_asr_lm_lm_train_lm_transformer2_en_bpe5000_valid.loss.ave_asr_model_valid.acc.ave/test_clean/logdir/keys.1.scp --asr_train_config exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/config.yaml --asr_model_file exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/valid.acc.ave_10best.pth --output_dir exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/decode_asr_lm_lm_train_lm_transformer2_en_bpe5000_valid.loss.ave_asr_model_valid.acc.ave/test_clean/logdir/output.1 --config conf/decode_asr.yaml --lm_train_config exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/lm/config.yaml --lm_file exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/lm/4epoch.pth 
# Started at Wed Feb  2 08:34:42 CST 2022
#
/home/xxxx/src/espnet/venv/bin/python3 /home/xxxx/src/espnet/espnet2/bin/asr_inference.py --batch_size 1 --ngpu 0 --data_path_and_name_and_type dump/raw/test_clean/wav.scp,speech,sound --key_file exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/decode_asr_lm_lm_train_lm_transformer2_en_bpe5000_valid.loss.ave_asr_model_valid.acc.ave/test_clean/logdir/keys.1.scp --asr_train_config exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/config.yaml --asr_model_file exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/valid.acc.ave_10best.pth --output_dir exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/decode_asr_lm_lm_train_lm_transformer2_en_bpe5000_valid.loss.ave_asr_model_valid.acc.ave/test_clean/logdir/output.1 --config conf/decode_asr.yaml --lm_train_config exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/lm/config.yaml --lm_file exp/espnet/simpleoier_librispeech_asr_train_asr_conformer7_hubert_ll60k_large_raw_en_bpe5000_sp/lm/4epoch.pth
2022-02-02 08:34:45,390 (asr:399) INFO: Vocabulary size: 5000
################################################################################
### WARNING, path does not exist: KALDI_ROOT=/mnt/matylda5/iveselyk/Tools/kaldi-trunk
###          (please add 'export KALDI_ROOT=<your_path>' in your $HOME/.profile)
###          (or run as: KALDI_ROOT=<your_path> python <your_script>.py)
################################################################################

Traceback (most recent call last):
  File "/usr/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/xxxx/src/espnet/espnet2/bin/asr_inference.py", line 606, in <module>
    main()
  File "/home/xxxx/src/espnet/espnet2/bin/asr_inference.py", line 602, in main
    inference(**kwargs)
  File "/home/xxxx/src/espnet/espnet2/bin/asr_inference.py", line 392, in inference
    speech2text = Speech2Text.from_pretrained(
  File "/home/xxxx/src/espnet/espnet2/bin/asr_inference.py", line 314, in from_pretrained
    return Speech2Text(**kwargs)
  File "/home/xxxx/src/espnet/espnet2/bin/asr_inference.py", line 85, in __init__
    asr_model, asr_train_args = ASRTask.build_model_from_file(
  File "/home/xxxx/src/espnet/espnet2/tasks/abs_task.py", line 1821, in build_model_from_file
    model = cls.build_model(args)
  File "/home/xxxx/src/espnet/espnet2/tasks/asr.py", line 405, in build_model
    frontend = frontend_class(**args.frontend_conf)
  File "/home/xxxx/src/espnet/espnet2/asr/frontend/s3prl.py", line 49, in __init__
    self.upstream, self.featurizer = self._get_upstream(frontend_conf)
  File "/home/xxxx/src/espnet/espnet2/asr/frontend/s3prl.py", line 70, in _get_upstream
    s3prl_upstream = torch.hub.load(
  File "/home/xxxx/src/espnet/venv/lib/python3.9/site-packages/torch/hub.py", line 399, in load
    model = _load_local(repo_or_dir, model, *args, **kwargs)
  File "/home/xxxx/src/espnet/venv/lib/python3.9/site-packages/torch/hub.py", line 425, in _load_local
    hub_module = import_module(MODULE_HUBCONF, hubconf_path)
  File "/home/xxxx/src/espnet/venv/lib/python3.9/site-packages/torch/hub.py", line 76, in import_module
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 855, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/home/xxxx/src/espnet/tools/s3prl/hubconf.py", line 1, in <module>
    from s3prl.hub import *
  File "/home/xxxx/src/espnet/tools/s3prl/s3prl/hub.py", line 30, in <module>
    _get_hubconf_entries()
  File "/home/xxxx/src/espnet/tools/s3prl/s3prl/hub.py", line 14, in _get_hubconf_entries
    _module = importlib.import_module(_module_name, package=__package__)
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/xxxx/src/espnet/tools/s3prl/s3prl/downstream/__init__.py", line 1, in <module>
    from . import experts
  File "/home/xxxx/src/espnet/tools/s3prl/s3prl/downstream/experts.py", line 20, in <module>
    _get_experts()
  File "/home/xxxx/src/espnet/tools/s3prl/s3prl/downstream/experts.py", line 11, in _get_experts
    _module = importlib.import_module(_module_name, package=__package__)
  File "/usr/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/xxxx/src/espnet/tools/s3prl/s3prl/downstream/quesst14_dtw/expert.py", line 11, in <module>
    from dtw import dtw
  File "/home/xxxx/src/espnet/venv/lib/python3.9/site-packages/dtw/__init__.py", line 17, in <module>
    from dtw.dtw import *
  File "/home/xxxx/src/espnet/venv/lib/python3.9/site-packages/dtw/dtw.py", line 36, in <module>
    from dtw._globalCostMatrix import _globalCostMatrix
  File "/home/xxxx/src/espnet/venv/lib/python3.9/site-packages/dtw/_globalCostMatrix.py", line 3, in <module>
    from dtw._dtw_utils import _computeCM_wrapper
  File "dtw/_dtw_utils.pyx", line 1, in init dtw._dtw_utils
ValueError: numpy.ndarray size changed, may indicate binary incompatibility. Expected 96 from C header, got 80 from PyObject
# Accounting: time=4 threads=1
# Ended (code 1) at Wed Feb  2 08:34:46 CST 2022, elapsed time 4 seconds

```

**Comment by sw005320** [comment]
Thanks for the report.
@simpleoier, could I ask you to check this issue?

**Comment by kamo-naoyuki** [comment]
I met the same error recently.

This is caused by importing `dtw` ( Numpy changed the ABI recently. Maybe, this issue was fixed by #3965 ? I'm not sure.)

Workaround:

- Uninstall `dtw` (`dtw` is not used in "upstream")
- Or, directly importing the upstream module, e.g. `s3prl.upstream.hubert.hubconf`, instead of  `s3prl.hubconf` 
- Or, using old numpy (I didn't try)

**Comment by ftshijt** [comment]
I have had similar issues recently as well. It's mostly due to the incompatible numpy version. I fixed that by upgrading numpy to >= 1.22
